views-react/src/main/java/io/micronaut/views/react/JSContextPool.java

```diff
@@ -0,0 +1,45 @@
+package io.micronaut.views.react;
+
+import io.micronaut.context.ApplicationContext;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import java.lang.ref.SoftReference;
+import java.util.LinkedList;
+
+/**
+ * Vends contexts to threads that need them.
+ */
+@Singleton
+class JSContextPool {
+    private final LinkedList<SoftReference<JSContext>> contexts = new LinkedList<>();
```

Comment:
you may want to consider a custom scope for `JSContext`

Also does a concurrent collection make more sense here? or is synchronised required? Perhaps a RentrantReadWriteLock would be more virtual thread friendly

Reply:
I narrowed the lock slightly but this is the best locking strategy:

- For Loom it's OK because we're not doing any blocking suspends whilst holding the lock.
- A concurrent collection could be used, but has more memory overhead and the lock should not be held for long given that it will normally only loop once or not at all. 

I've profiled this code already and it didn't show up. The overhead of executing the JS drowns everything else.

Reply:
W.R.T the custom scope, I'm not sure how to implement pooling that way or what benefit it would give. Creating these via the bean context seems easy enough. Is there IoC container support for soft object pooling I could use instead?

Reply:
ok it is probably fine as is